src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -804,5 +814,96 @@ public void Dispose()
+            {
+                _operation = operation;
+                _id = id;
+            }
+
+            public void Store(string name, Stream stream, string contentType = null)
+            {
+                _operation._attachmentsOperation.Store(_id, name, stream, contentType);
+            }
+
+            public Task StoreAsync(string name, Stream stream, string contentType = null, CancellationToken token = default)
+            {
+                return _operation._attachmentsOperation.StoreAsync(_id, name, stream, contentType, token);
+            }
+        }
+
+        public class AttachmentsBulkInsertOperation
+        {
+            private readonly BulkInsertOperation _operation;
+            private readonly CancellationToken _token;
+
+            public AttachmentsBulkInsertOperation(BulkInsertOperation operation)
+            {
+                _operation = operation;
+                _token = _operation._token;
+            }
+
+            public void Store(string id, string name, Stream stream, string contentType = null)
+            {
+                AsyncHelpers.RunSync(() => StoreAsync(id, name, stream, contentType, token: default));
+            }
+
+            public async Task StoreAsync(string id, string name, Stream stream, string contentType = null, CancellationToken token = default)
+            {
+                PutAttachmentCommandHelper.ValidateStream(stream);
+
+                using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(token, _token);
+                using (_operation.ConcurrencyCheck())
+                {
+
+                    switch (_operation._inProgressCommand)
+                    {
+                        case CommandType.Counters:
+                            _operation._countersOperation.EndPreviousCommandIfNeeded();
+                            break;
+                        case CommandType.TimeSeries:
+                            TimeSeriesBulkInsert.ThrowAlreadyRunningTimeSeries();
+                            break;
+                    }
+
+                    await _operation.ExecuteBeforeStore().ConfigureAwait(false);
+
+                    try
+                    {
+                        if (_operation._first == false)
+                            _operation._currentWriter.Write(",");
+
+                        _operation._currentWriter.Write("{\"Id\":\"");
+                        WriteString(_operation._currentWriter, id);
+                        _operation._currentWriter.Write("\",\"Type\":\"AttachmentPUT\",\"Name\":\"");
+                        WriteString(_operation._currentWriter, name);
+                        if (contentType != null)
+                        {
+                            _operation._currentWriter.Write("\",\"ContentType\":\"");
+                            WriteString(_operation._currentWriter, contentType);
+                        }
+                        _operation._currentWriter.Write("\",\"RavenBlobSize\":");
```

Comment:
If we are not wrapping inside the structure like in Smuggler, then call this just 'ContentLength'. It will match other property 'ContentType' nicely.